src/kakarot/accounts/contract/library.cairo

```diff
@@ -56,6 +67,53 @@ namespace ContractAccount {
         return ();
     }
 
+    // @notice This function is a factory to handle salt and registration of EVM<>Starknet binding.
+    // @param salt: The salt for computing the corresponding EVM address
+    func deploy{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr,
+        bitwise_ptr: BitwiseBuiltin*,
+    }(salt: felt) -> (evm_contract_address: felt, starknet_contract_address: felt) {
+        alloc_locals;
+
+        // Prepare constructor data
+        let (local calldata: felt*) = alloc();
+        let (kakarot_address) = get_contract_address();
+        assert [calldata] = kakarot_address;
+        assert [calldata + 1] = 0;
+
+        // Deploy contract account with no bytecode
+        let (class_hash) = evm_contract_class_hash.read();
+        let (starknet_contract_address) = deploy_syscall(
+            class_hash=class_hash,
+            contract_address_salt=salt,
+            constructor_calldata_size=2,
+            constructor_calldata=calldata,
+            deploy_from_zero=0,
```

Comment:
does that mean we're deploying with `msg.sender = addres(0)`

Reply:
Well, this was like that already, it's 0 for FALSE, I've put FALSE again